### ISSUE TITLE
Include full URLs in search results

### ISSUE THREAD

**Issue by kauffj** [lbryio/lighthouse#31]
URLs are typically what will ultimately be resolved or used by a service or application utilizing the search API. So it'd probably make sense to just return these.

**Comment by tiger5226** [comment]
@kauffj I wanted to just get this one done, however, there could be dependencies, and there may be alternative uses of getting claim value details based on a search result. Either way it returns all that it does because the data is stored in elastic search as an elastic document that can be summed up as a json object of information. The results are for these documents. If we wanted to return the URLs it would just be some minor post processing.

Since there are external dependencies on this API it is probably best to create a new API:

current http://localhost:50005/search?s=<term>
new http://localhost:50005/url/search?s=<term>

This way we can have the service live before it is used live by the app and the current one can still be used unhindered until the adjustment in the app is ready. Also it reduces golive dependencies. The drawback is now we have 2 apis.

**Comment by kauffj** [comment]
I was going through tickets assigned to me and realized I never saw above until now. Good catch of a real issue, but I'd propose the proper fix for this is likely to add a version parameter the endpoint to allow graceful migration, not diverge endpoints.

**Comment by tiger5226** [comment]
Yeah that makes sense! Will do

**Comment by tiger5226** [comment]
@seanyesmunt we have the PR ready, but I just wanted to double check with you that we are not using the search APIs anywhere else.

**Comment by seanyesmunt** [comment]
They are used in the android app. It would definitely be a smoother transition to include versioning in the api

**Comment by kauffj** [comment]
I'm not 100% on what I meant by this, but I _think_ I just meant to actually include the URL so it doesn't have to be reconstructed, not to actually strip out the metadata (though that may be fine).

Ideally, it's return [the canonical url](https://github.com/lbryio/lbry/issues/958).

**Comment by kauffj** [comment]
It's still not performant to display/determine the shortest possible canonical URL, but I'm fine waiting on the daemon to fix that. 

That said, even without proper canonical URLs, I do think it makes sense to return the full URL. It'd be like if a search engine returned the host and TLD separately (technically host includes TLD but you get what I mean).

**Comment by seanyesmunt** [comment]
@tiger5226 Whoops. You are right. I thought we were handling that data differently.

**Comment by tiger5226** [comment]
All clear. I will leave the current fields alone and add one for the canonical url which should be:
`<claim_name>#<claim_id>` is this correct @kauffj ? 

I reviewed the issue linked, but I don't see what exactly we want the canonical url to be. Sorry if I missed something.

**Comment by kauffj** [comment]
@tiger5226 no, the canonical URL is more complex than that. Given full access to chainquery however, it should be possible to compute it in a single query/pass.

I updated https://github.com/lbryio/lbry/issues/958 to (hopefully) be significantly more clear

**Comment by tiger5226** [comment]
@kauffj This seems very complex. In a given query each row would have to compare against all other rows with a similar set of fields. Do you think this is something I should adjust Chainquery to calculate when a new claim comes in? Also what is the impact if a claim is abandoned? I presume that doesn't matter. If it existed in the blockchain at some point it impacts the canonical url. Is this correct? Adjusting Chainquery seems like a good idea. I could see someone using the API to gain quick access to the canonical URL or getting the claimid from a canonical url. If chainquery has already done these calculations it will be much faster. Also the adjustment to the lighthouse query becomes super simple as well. We just add a field to grab and push to elastic search.

**Comment by kauffj** [comment]
@tiger5226 canonical URLs cannot (and won't) change even if a claim is abandoned.

We can probably wait to handle this in lighthouse until it's been added at lower levels.

**Comment by tiger5226** [comment]
ok I will put this in the icebox.

**Comment by alyssaoc** [comment]
@tiger5226 Is this ready for review or going to the icebox?

**Comment by tiger5226** [comment]
icebox, corrected the pipeline to triaged.

**Comment by lyoshenka** [comment]
closing until it's been added at a lower level
